### PR TITLE
Fix ActionTesting::InitializeDatabox for when a tag is removed

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This previously failed to compile when one of the tags was removed during initialization.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/ActionTesting.hpp [hunk 1]
```diff
@@ -187,15 +187,18 @@ struct InitializeDataBox<tmpl::list<SimpleTags...>, ComputeTagsList> {
...
   static std::tuple<db::DataBox<DbTagsList>&&> apply(
-      db::DataBox<DbTagsList>& box,
+      db::DataBox<DbTagsList>& /*box*/,
       const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
       const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
       const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
       const ParallelComponent* const /*meta*/) noexcept {
-    return {std::move(box)};
+    ERROR(
+        "Tried to apply ActionTesting::InitializeDatabox even though one or "
```
Comment (moxcodes) [inline tests/Unit/ActionTesting.hpp]: `InitializeDataBox` (caps on B)

**Review by moxcodes — DISMISSED** [review]
Also "that that" in commit message. Only small nitpicks for this. group code review with @fmahebert and @markscheel .

**Comment by nilsvu** [comment]
@moxcodes squashed in your comment.

BTW when doing a group review please mark one of the reviewers as "main" or "primary" or something, so that it's clear who will follow up on fixups and get the PR merged. I'm assuming it's you @moxcodes since you posted the comment :)

@nilsdeppe could you look at this when you get the chance?

**Review by moxcodes — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]
Travis failure is reported as bug in #1663 (random failure in StrahlkorperGr.AreaElement)

**Pull request merged**